src/__tests__/__views__/CreateEditArticle.test.js

```diff
@@ -0,0 +1,231 @@
+import React from "react";
+import { shallow, mount } from "enzyme";
+import {
+  Article,
+  mapDispatchToProps,
+  mapStateToProps
+} from "../../views/CreateEditHolder";
+import { article, article1, article2, article3 } from "../__mocks__/testData";
+
+const shallowSetup = (pathname = "/articles/new", articleFields) => {
+  const props = {
+    article: {
+      ...articleFields
+    },
+    history: {
+      push: jest.fn(),
+      location: {
+        pathname
+      }
+    },
+    match: {
+      params: {
+        slug: "hello-world"
+      }
+    },
+    response: "",
+    fetchArticleToEdit: jest.fn(),
+    createArticle: jest.fn(),
+    editOneArticle: jest.fn()
+  };
+  const wrapper = shallow(<Article {...props} />);
+  return { wrapper, props };
+};
+
+const mountSetup = (pathname, articleFields) => {
+  const props = {
+    article: {
+      ...articleFields
+    },
+    history: {
+      push: jest.fn(),
+      location: {
+        pathname
+      }
+    },
+    match: {
+      params: {
+        slug: "hello-world"
+      }
+    },
+    response: "",
+    fetchArticleToEdit: jest.fn(),
+    createArticle: jest.fn(),
+    editOneArticle: jest.fn()
+  };
+  const wrapper = mount(<Article {...props} />);
+  return { wrapper, props };
+};
+
+describe("Create/Edit article", () => {
+  test("should render the creating article interface", () => {
+    const { wrapper } = shallowSetup(undefined, article);
+    expect(wrapper.find(".cancel")).toHaveLength(0);
+  });
+
+  test("should call oninput change method ", () => {
+    const { wrapper } = shallowSetup(undefined, article);
+    wrapper.find(".title__input").simulate("change", {
+      target: { value: "hello world", name: "title" }
+    });
+    expect(wrapper.state().title).toEqual("hello world");
+  });
+
+  test("should call create article method with correct input ", () => {
+    const { wrapper, props } = shallowSetup(undefined, article);
+    wrapper.find(".title__input").simulate("change", {
+      target: { value: "hello world", name: "title" }
+    });
+    wrapper.instance().onEditorChange(article.body);
+    wrapper.find(".description__input").simulate("change", {
+      target: { value: "hello world", name: "description" }
+    });
+    wrapper.find(".publish").simulate("click");
+    expect(props.createArticle).toHaveBeenCalled();
+  });
+  test("should cancel edit ", () => {
+    const { wrapper, props } = mountSetup(
+      "/articles/helloWorld/edit",
+      article3
+    );
+    wrapper.setProps(article);
+    wrapper.instance().cancelEdit();
+    expect(props.history.push).toHaveBeenCalled();
```

Comment:
What are you testing here?
Is it a redirection?
If yes it may be better to use toHaveBeenCalledwith to check if history.push was called with URL you are redirecting to.